check-ssl-cert/lib/check-ssl-cert.go

```diff
@@ -63,8 +67,29 @@ func run(args []string) *checkers.Checker {
 	return checkers.NewChecker(chkSt, msg)
 }
 
-func getCert(addr string) (*x509.Certificate, error) {
-	conn, err := tls.Dial("tcp", addr, &tls.Config{})
+func getCert(addr string, opts *certOpts) (*x509.Certificate, error) {
+	tlsConfig := &tls.Config{}
+	if opts.CAfile != "" {
+		caCert, err := os.ReadFile(opts.CAfile)
+		if err != nil {
+			return nil, err
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+		tlsConfig.RootCAs = caCertPool
+		tlsConfig.BuildNameToCertificate()
```

Comment:
[may] Today I noticed this method is marked as deprecated.
https://pkg.go.dev/crypto/tls#Config.BuildNameToCertificate